feed-crawler/test/claude-ai.e2e-spec.ts

```diff
@@ -0,0 +1,164 @@
+import { ResultSetHeader } from 'mysql2';
+import { ClaudeService } from '../src/claude.service';
+import { setupTestContainer } from './setup/testContext.setup';
+
+describe('Claude AI e2e-test', () => {
+  const testContext = setupTestContainer();
+  let claudeService: ClaudeService, feedData;
+  const feedRedisAiQueueData: any = {
+    content: 'test',
+    deathCount: 0,
+  };
+
+  beforeAll(async () => {
+    claudeService = new ClaudeService(
+      testContext.tagMapRepository,
+      testContext.feedRepository,
+      testContext.redisConnection,
+    );
+
+    const rssData = (await testContext.dbConnection.executeQuery(
+      `INSERT INTO rss_accept (name, user_name, email, rss_url, blog_platform) VALUES (?, ?, ?, ?, ?)`,
+      ['test', 'test_name', 'test@test.com', 'https://test.com/rss', 'etc'],
+    )) as any as ResultSetHeader;
+
+    feedData = (await testContext.dbConnection.executeQuery(
+      `INSERT INTO feed (created_at, title, path, thumbnail, blog_id) VALUES (?, ?, ?, ?, ?)
+      `,
+      [new Date(), 'test', 'test', 'test', rssData.insertId],
+    )) as any;
+
+    await testContext.dbConnection.executeQuery(
+      `INSERT INTO tag (name) VALUES (?), (?), (?)`,
+      ['test1', 'test2', 'test3'],
+    );
+  });
+
+  it('피드의 데이터를 요약하고 태그를 받아왔을 때, 태그가 DB 목록에 없다.', async () => {
+    // given
+    jest
+      .spyOn(claudeService as any, 'loadFeeds')
+      .mockResolvedValue([feedRedisAiQueueData]);
+
+    jest.spyOn(claudeService as any, 'requestAI').mockResolvedValue({
+      ...feedRedisAiQueueData,
+      id: feedData.insertId,
+      summary: 'test summary',
+      tagList: ['test4'],
+    });
+
+    // when
+    await claudeService.startRequestAI();
+
+    // then
+    const [searchSummary] = await testContext.dbConnection.executeQuery(
+      `SELECT * FROM feed WHERE feed.id = ?`,
+      [feedData.insertId],
+    );
+    const tagList = await testContext.dbConnection.executeQuery(
+      `SELECT name FROM tag, tag_map WHERE tag.id = tag_map.tag_id AND tag_map.feed_id = ?`,
+      [feedData.insertId],
+    );
+
+    expect(searchSummary['summary']).toStrictEqual(null);
+    expect(tagList.map((t: any) => t.name)).toStrictEqual([]);
+  });
```

Comment:
P4: 본 방식은 `모종의 이유로 삽입에 실패할 경우 데이터 업데이트가 수행되지 않아 빈 값이 존재한다.` 라는 행위로 검증을 수행하고 있습니다.

문득 보다가, **테스트에서 “빈 결과”를 검증하기보다, 특정 상황에서는 이러한 에러가 발생한다**는 걸 명시적으로 검증하는 게 더 깔끔 하지 않을까? 라고 생각했어요. (기존에 서버측 테스트들은 e2e에 가까워서 내부적인 에러를 잡아내기 적절하지 않았지만요!)

[참고](https://jestjs.io/docs/expect#rejects)
```suggestion
    // when-then
  await expect(claudeService.startRequestAI())
    .rejects
    .toThrow(
      `${feedData.insertId}번 피드 AI 요약 결과에 존재하지 않는 태그 가 포함되어 있습니다.`
    );
```

문자열로 에러내용을 검증하는게 조금 걸리기는 한데,,, _(더 좋은 방법은 커스텀 에러를 만든 다음 해당 에러타입으로 검증하는 방법이 있을 것 같긴 한데, 이건 좀 귀찮으니까요 🤣)_ 어떻게 생각하시나요??

Reply:
성윤님 코드가 더 나은 것 같네요! 변경하겠습니다